Program.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Diagnostics;
+
+namespace ConsoleApp2
+{
+    class Program
+    {
+        
+        static void Main(string[] args)
+        {
+            Segregator segregator = new Segregator();
+            List<int> data = new List<int>() {3,4,5,6,8,7,8,3,13,14,15,16,17,18,13,22,23};
```

Comment:
Test data needs to have variations. For example single value {2}, repeated values {4,4,4}, no value list, so that boundary values are tested too.